src/full_node/full_node.py

```diff
@@ -287,6 +293,26 @@ async def _on_connect(self) -> OutboundMessageGenerator:
         async for msg in self._send_tips_to_farmers(Delivery.RESPOND):
             yield msg
 
+    async def periodically_peer_gossip(self):
+        while True:
+            # Randomly choose to get peers from 12 to 24 hours.
+            sleep_interval = random.randint(3600 * 12, 3600 * 24)
```

Comment:
Generally speaking, don't use non-deterministic global functions like `random.randint` or `time.time`; take an additional parameter instead (`random_stream` which could just yield a bunch of values, or `now`). This makes deterministic testing possible.